tools/analysis_tools/visualization_cam.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+"""Use the pytorch-grad-cam tool to visualize Class Activation Maps (CAM).
+
+requirement: pip install grad-cam
+"""
+
+from argparse import ArgumentParser
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from mmengine.model import revert_sync_batchnorm
+from PIL import Image
+from pytorch_grad_cam import GradCAM, LayerCAM, XGradCAM, GradCAMPlusPlus, EigenCAM, EigenGradCAM
+from pytorch_grad_cam.utils.image import preprocess_image, show_cam_on_image
+
+from mmseg.apis import inference_model, init_model, show_result_pyplot
+from mmseg.utils import register_all_modules
+
+
+class SemanticSegmentationTarget:
+    """wrap the model.
+
+    requirement: pip install grad-cam
+
+    Args:
+        category (int): Visualization class.
+        mask (ndarray): Mask of class.
+        size (tuple): Image size.
+    """
+
+    def __init__(self, category, mask, size):
+        self.category = category
+        self.mask = torch.from_numpy(mask)
+        self.size = size
+        if torch.cuda.is_available():
+            self.mask = self.mask.cuda()
+
+    def __call__(self, model_output):
+        model_output = torch.unsqueeze(model_output, dim=0)
+        model_output = F.interpolate(
+            model_output, size=self.size, mode='bilinear')
+        model_output = torch.squeeze(model_output, dim=0)
+
+        return (model_output[self.category, :, :] * self.mask).sum()
+
+
+def main():
+    parser = ArgumentParser()
+    default_cfg = 'configs/deeplabv3/deeplabv3_r50-d8_4xb2-40k_cityscapes-769x769.py'  # noqa
+    parser.add_argument('--img', default='demo/demo.png', help='Image file')
+    parser.add_argument('--config', default=default_cfg, help='Config file')
+    parser.add_argument(
+        '--checkpoint',
+        default='deeplabv3_r50-d8_769x769_40k_cityscapes.pth',
+        help='Checkpoint file')
```

Comment:
It is recommended to force the user to enter `img`, `config` and `checkpoint`.

Reply:
Now the user need to enter  img, config and checkpoint.